drivers/wifi/wfx200/wfx200_host.c

```diff
@@ -0,0 +1,730 @@
+/**
+ * Copyright (c) 2022 grandcentrix GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_MODULE_NAME wifi_wfx200_host
+#define LOG_LEVEL CONFIG_WIFI_LOG_LEVEL
+
+#include <sl_wfx.h>
+#undef BIT
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <stdarg.h>
+#include <stdint.h>
+
+#include <zephyr.h>
+#include <device.h>
+#include <sys/util.h>
+#include <drivers/spi.h>
+#include <drivers/gpio.h>
+
+#include "wfx200_internal.h"
+
+#include <sl_wfx_wf200_C0.h>
+
+/* The driver from SiLabs does use a internal global variable to store the
+ * drivers context. This means, that we are limitied to one instance of SiLabs
+ * driver and need to use a global variable here as well, as the host api does
+ * not give a context to the callbacks. So we need to use our own global
+ * variable here as well.
+ */
+extern struct wfx200_data *wfx200_0;
+
+/* WFX host callbacks */
+void sl_wfx_connect_callback(sl_wfx_connect_ind_t *connect);
+void sl_wfx_disconnect_callback(sl_wfx_disconnect_ind_t *disconnect);
+void sl_wfx_start_ap_callback(sl_wfx_start_ap_ind_t *start_ap);
+void sl_wfx_stop_ap_callback(sl_wfx_stop_ap_ind_t *stop_ap);
+void sl_wfx_host_received_frame_callback(sl_wfx_received_ind_t *rx_buffer);
+void sl_wfx_scan_result_callback(sl_wfx_scan_result_ind_t *scan_result);
+void sl_wfx_scan_complete_callback(sl_wfx_scan_complete_ind_t *scan_complete);
+void sl_wfx_generic_status_callback(sl_wfx_generic_ind_t *frame);
+void sl_wfx_ap_client_connected_callback(sl_wfx_ap_client_connected_ind_t *ap_client_connected);
+void sl_wfx_ap_client_rejected_callback(sl_wfx_ap_client_rejected_ind_t *ap_client_rejected);
+void sl_wfx_ap_client_disconnected_callback(sl_wfx_ap_client_disconnected_ind_t *ap_client_disconnected);
+void sl_wfx_ext_auth_callback(sl_wfx_ext_auth_ind_t *ext_auth_indication);
+
+/**
+ * Functions for initializing the host, firmware uploading and pds
+ */
+sl_status_t sl_wfx_host_init(void)
+{
+	/* Initialization happens in wfx200_init, nothing to do here */
+	return SL_STATUS_OK;
+}
+
+sl_status_t sl_wfx_host_get_firmware_data(const uint8_t **data, uint32_t data_size)
+{
+	if (data == NULL || wfx200_0->firmware_pos >= sizeof(sl_wfx_firmware)) {
```

Comment:
It would make sense to expose this to the application, because various applications might require getting Wi-Fi firmware from different sources. 

E.g. we didn't have enough flash and had to load Wi-Fi firmware from external memory (in our case, SD card).

I might imagine that other applications might require other ways to update it (e.g. via networks).

Reply:
It totally would make sense. As the firmware can be 350 kB big, there should be an alternative way to provide the firmware instead of just link it in. I just didn't do this in this pr, because i didn't want to make it bigger as it already is and rather make a separate pr for adding this. If you think, this would be needed to make any practical use of this driver, i'll add a device specific api for this.

Reply:
Yes, it would be extremely nice to have that API, otherwise to use this driver we will have to wait for yet another PR to be merged...